libs/authentication/api/src/lib/optional-authentication.guard.ts

```diff
@@ -1,16 +1,10 @@
-import {AuthGuard} from '@nestjs/passport'
 import {ExecutionContext, UnauthorizedException} from '@nestjs/common'
-import {GqlExecutionContext} from '@nestjs/graphql'
+import {AuthenticationGuard} from './authentication.guard'
 
 /**
  * This Guard is to attach the user to the request
  */
-export class OptionalAuthenticationGuard extends AuthGuard('session') {
-  public override getRequest(context: ExecutionContext) {
-    const ctx = GqlExecutionContext.create(context)
-    return ctx.getContext().req
-  }
-
+export class OptionalAuthenticationGuard extends AuthenticationGuard {
```

Comment:
Don't think this should extend

Reply:
Yeah but the logic around them still isn't the same, if `AuthenticationGuard` now gets some more authentication features, this one will also get them which shouldn't.

While they have some similar code, they are not the same thing logically.